src/librustdoc/html/static/main.js

```diff
@@ -2257,9 +2257,9 @@ function defocusSearchBar() {
 
         function implHider(addOrRemove, fullHide) {
             return function(n) {
-                var is_method = hasClass(n, "method") || fullHide;
-                if (is_method || hasClass(n, "type")) {
-                    if (is_method === true) {
+                var shouldHide = hasClass(n, "method") || hasClass(n, "associatedconstant");
+                if (shouldHide === true || fullHide === true || hasClass(n, "type")) {
+                    if (shouldHide === true || fullHide === true) {
```

Comment:
Why did you move out the check to `fullHide` here? There's no time you'd want to treat it separately from the method checks, right?

Reply:
Well, in the case it's an associated constant. ;)

Reply:
I don't understand what this means. I'm asking why `shouldHide` is treated separately from `fullHide`.

Reply:
I did noticed this part but both looks the same to me, either way looks fine but I prefer `fullHide` being part of `shouldHide`.

Reply:
It's a bit different: we don't hide "type" elements, only their doc block. So the first condition is either we should completely hide it or its doc block, the second one is if we should hide the element only.

Reply:
@GuillaumeGomez that's not my question. My question is why did you change from

```rust
var shouldHide = hasClass(n, "method") || hasClass(n, "associatedconstant") || fullHide; 
if (shouldHide) { ... }
```
to
```rust
var shouldHide = hasClass(n, "method") || hasClass(n, "associatedconstant"); 
if (shouldHide || fullHide) { ... }
```
The second is much easier to forget the check to `fullHide`, and there's no need to make it separate.

Reply:
I thought on the contrary that it made the code easier to follow but I can put back `fullHide` into the variable.

Reply:
That would be great, thanks :)